bigquery/google/cloud/bigquery/client.py

```diff
@@ -1529,19 +1531,24 @@ def insert_rows(self, table, rows, selected_fields=None, **kwargs):
         Raises:
             ValueError: if table's schema is not set
         """
-        if isinstance(table, str):
-            table = TableReference.from_string(table, default_project=self.project)
+        table = _table_arg_to_table(table, default_project=self.project)
+
+        if not isinstance(table, Table):
```

Comment:
Would it make sense to put this in `_table_arg_to_table()`?

Reply:
Maybe, but I'd prefer to keep it consistent with `_table_arg_to_table_ref` and if we throw there we'd have to then catch the exception in a few spots, which I think is asking for trouble and less in line with Python's "duck typing" philosophy.

Reply:
Fair - I'm probably too type strict for python :D